File: creaking-coding-interview/twoStrings.cpp
//
//  main.cpp
//  hello_world
//
//  Created by Shekhar Suman on 06/01/20.
//  Copyright © 2020 Shekhar Suman. All rights reserved.
//

#include <iostream>
#include <map>
using namespace std;

string twoStrings(string s1, string s2) {
    map<char, bool> map1;
    map<char, bool> map2;
    for(int i = 0; i < s1.length(); i++) {
        map1.insert(pair<char, bool>(s1.at(i), true));
    }
    for(int i = 0; i < s2.length(); i++) {
        map2.insert(pair<char, bool>(s2.at(i), true));
    }
    map<char, bool>::iterator it = map1.begin();
    for(pair<char, bool> element: map1) {
        if(map2.count(element.first)){
            return "YES";
        }
    }
    return "NO";
}

int main(int argc, const char * argv[]) {
    // insert code here...
    string result = twoStrings("hello", "ol");
    cout<<result;
    return 0;
}

File: creaking-coding-interview/twoStrings.go
package interview

import "fmt"

func twoStrings(s1 string, s2 string) string {
	var map1 = make(map[byte]bool)
	var map2 = make(map[byte]bool)
	for i := 0; i < len(s1); i++ {
		map1[s1[i]] = true
	}
	for i := 0; i < len(s2); i++ {
		map2[s2[i]] = true
	}
	for key := range map1 {
		if map2[key] {
			return "YES"
		}
	}
	return "NO"
}

func main() {
	fmt.Println(twoStrings("Hello", "ol"))
}
